File: core/src/de/project/ice/ecs/components/ScriptComponent.java
package de.project.ice.ecs.components;

import de.project.ice.scripting.Script;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ScriptComponent implements IceComponent<ScriptComponent>
{
    @Nullable
    public Script script = null;
    @NotNull
    public String scriptName = "";

    public void setScriptName(String scriptName)
    {
        scriptName = scriptName;
        script = null;
    }

    @Override
    public void reset()
    {
        script = null;
        scriptName = "";
    }

    @Override
    public void copyTo(ScriptComponent copy)
    {
        copy.scriptName = scriptName;
    }
}

File: core/src/de/project/ice/ecs/components/ScriptComponent.kt
package de.project.ice.ecs.components

import de.project.ice.annotations.Property
import de.project.ice.scripting.Script

class ScriptComponent : CopyableIceComponent {
    @Property("The name if the Script (Classname)")
    var script: Script? = null
    var scriptName = ""
        set(scriptName) {
            field = scriptName
            script = null
        }

    override fun reset() {
        script = null
        scriptName = ""
    }

    override fun copyTo(copy: CopyableIceComponent) {
        if (copy is ScriptComponent) {
            copy.scriptName = scriptName
        }
    }
}
